### ISSUE TITLE
[mv][testbed] Executing Lola

### ISSUE THREAD

**Issue by gulyx** [LearnPAd/learnpad#388]
@damianofalcioni, @imatesiu, @woshilapin, @semmeneg, @tomjorquera @acalabro 
- lola is the analyzer "generated" and executed by the MV component 
- lola will always be located in /tmp
- the current version of lola is classified as follows  (by the 'file' command):
  `lola: ELF 32-bit LSB  executable, Intel 80386, version 1 (GNU/Linux), dynamically linked (uses shared libs), for GNU/Linux 2.6.24, BuildID[sha1]=894bb774ae645f80b983900e304432bf7ebf2087, not stripped`

in conclusion the server hosting Learn PAd must have the 32-bit compatibility libraries installed.

not sure the http://testbed.learnpad.eu currently has this libraries installed.

**Comment by gulyx** [comment]
@woshilapin did u checked if the testbed has the 32-bit compatibility libs installed?

**Comment by woshilapin** [comment]
I've tried to deploy the new version of the platform and reach a few problems with this `lola` 32 bits binary.

I've documented a part [here](https://github.com/LearnPAd/learnpad/blob/master/lp-platform/DEPLOYMENT.md#notes-about-mv-component).

To be precise, I'll detail here some of the procedures I've done.

First of all, identify what is needed to run `lola`.

```
$> readelf -d /tmp/lola | grep NEEDED
 0x00000001 (NEEDED)                     Shared library: [libstdc++.so.6]
 0x00000001 (NEEDED)                     Shared library: [libm.so.6]
 0x00000001 (NEEDED)                     Shared library: [libgcc_s.so.1]
 0x00000001 (NEEDED)                     Shared library: [libpthread.so.0]
 0x00000001 (NEEDED)                     Shared library: [libc.so.6]
```

Therefore, I tried to install the following dependencies, for an `i386` architecture.

```
dpkg --add-architecture i386
apt-get update
apt-get install libstdc++6:i386 libc6:i386
```

However, I'm on Debian Wheezy (this is our actual server infrastructure so I don't have a choice) and if I try to run `lola`, this is what I get.

```

**Comment by gulyx** [comment]
@damianofalcioni any "elegant" and alternative idea on how to solve this problem with Lola?

**Comment by damianofalcioni** [comment]
@gulyx @woshilapin The best solution i guess will be to recompile the sources on the server and update the linux bin in github with the new one

**Comment by woshilapin** [comment]
On the server may be the best but having already a 64-bits binary with embedded libraries should already be enough.  Could you do that @damianofalcioni?

**Comment by damianofalcioni** [comment]
@woshilapin @gulyx I've just created a pull request

**Comment by woshilapin** [comment]
Thanks Damiano, as explained in the PR #423, I tested it but it doesn't work for obvious reasons.  When you provide such a solution (an external binary which is not built as part of the platform), these the kind of caveats you always reach.  At the very least, the binary should be self-sufficient with no dependencies on external libraries (they should be embedded).

Another solution would be that you include Lola in the build.  Due to our custom build system which is not full-Java, it's possible to do.  You just have to modify the `build` script inside your component.

**Comment by cjdelisle** [comment]
here's the source for lola http://download.gna.org/service-tech/lola/lola-2.0.tar.gz 
basically we need to just invoke a little script which does:
```untar && cd lola2 && ./configure && ./make && cp ./src/lola /tmp/```
should be pretty trivial, as long as it builds on the target systems we want.

**Comment by cjdelisle** [comment]
One trickey issue is that you can't just dump lola into ```/tmp```, java decided on it's own tmpdir so you need to really put it in the source code while building so that it will use the right lola when it is putting the file into the tmpdir
